src/service_info.rs

```diff
@@ -366,6 +366,18 @@ impl ServiceInfo {
             .cloned()
             .unwrap_or(ServiceStatus::Unknown)
     }
+
+    /// Returns a resolved service instance.
+    pub fn as_resolved_service(&self) -> ResolvedService {
+        ResolvedService {
+            ty_domain: self.ty_domain.clone(),
+            fullname: self.fullname.clone(),
+            host: self.server.clone(),
+            port: self.port,
+            addresses: self.addresses.clone(),
+            txt_properties: self.txt_properties.clone(),
```

Comment:
To be consistent with `ServiceInfo`, I am cloning `TxtProperties`,  not creating a new `HashMap` of entries inside `TxtProperties` (see issue #299).  We can handle data extraction of `TxtProperties` separately.